apps/full-site-editing/full-site-editing-plugin/starter-page-templates/page-template-modal/index.js

```diff
@@ -279,15 +294,39 @@ if ( tracksUserData ) {
 	initializeWithIdentity( tracksUserData );
 }
 
-registerPlugin( 'page-templates', {
+// Open plugin only if we are creating new page.
+if ( screen.action === 'add' ) {
+	registerPlugin( 'page-templates', {
+		render: () => {
+			return (
+				<PageTemplatesPlugin
```

Comment:
Would this theoretically mean that there are two instances of the page template selector when creating a page? i.e. first from adding, and then it is hidden, and then another from the sidebar? I'm wondering if there's a way to avoid that. Maybe we can default the modal to "open" from the sidebar if action === add?

Reply:
The one in the sidebar is never called until the button is clicked and the warning modal is confirmed.  AFAIK its instance does not exist until that prop is changed and the component is called and mounts.  Once the modal called from the sidebar is then closed, that switch `toggleTemplateModal` prop we passed in ends the instance of the component altogether.

However, the way this template selector is initially set up (when it loads by default on creating a new page), the way it closes only bypasses the render function, meaning that the instance never closes.  So if one were to create a new page and apply a template there would still be 1 instance active, when they open it through the sidebar the second instance will be active, but when they close that one that was triggered from the sidebar the second instance will be inactive again.  So there is some period of time in that flow that there are 2 instances going, and it is a result of how the initial modal was designed to 'close'.

Reply:
Cool :)